### ISSUE TITLE
As a Mina user, I want to read documentation and articles about Mina and Zero-knowledge proofs

### ISSUE THREAD

**Issue by JanSlobodnik** [openmina/openmina#117]
### 13 January 2025
* [What is the Web Node and how does it work Article Series](https://docs.google.com/document/d/1iPZHprMT5hV37rAm88PUwrl8FG51HcT1LXWz2FwMWys/edit?usp=sharing)

### 13 January 2025
* started working on zkTLS article
* finished zk-noid, waiting for final approval, legal review, then will publish this week
* e-grains is finished, needs image and then scheduling for early/mid february

### 6 January 2025

* finished text part of the [e-grains article](https://docs.google.com/document/d/1yZk2uMy2SoY63E0szFtpNrgcoaRhA8T30k7zfXjE6Ys/edit?usp=sharing). its ready for reading / legal review
* nearly finished with [ZK noid article](https://docs.google.com/document/d/1Ul43wgPXs0Y0TBI6njDh--kAdPy6oyccecxIINLV0gc/edit?usp=sharing), waiting for quote.
* Working on biosnarks article but would like to also support Lukas with Web Node GTM

### 9 December
* [ZK Noid (ZK gaming) use case first draft](https://docs.google.com/document/d/1Ul43wgPXs0Y0TBI6njDh--kAdPy6oyccecxIINLV0gc/edit?usp=sharing)
* While I wait on feedback I am focusing on the [BioSNARKs use case](https://docs.google.com/document/d/10ltciATXpukmr2qUSV3nd8rfXJA3fHli8Gkar076byU/edit?usp=sharing)
* e-grains use case is nearly finished but it is shelved until we get a response from their team regarding the IoT device/oracle.

### 2 December
Last week I was OOO, now I've resumed work to the [ZKON e-grains case study](https://docs.google.com/document/d/1yZk2uMy2SoY63E0szFtpNrgcoaRhA8T30k7zfXjE6Ys/edit?usp=sharing)

### 18 November
* [Marketing framework](https://docs.google.com/document/d/1_UEZ5sxvzekJt02f2I5P50xNYf80X6NYTL_E-1bfvjM/edit?usp=sharing) for Mina Web Node launch

### 11 November
* Page for ZK app devs to decide whether they use [protokit or o1js.](https://github.com/JanSlobodnik/docs2/blob/471578a82f7fadf47be1e3679c43585cd3e13d6c/docs/zkapps/zkapp-development-frameworks.mdx), im waiting for a diagram from Lucian before I make a pull request on the main docs website.
* working on the redesign, [first version](https://docs.google.com/document/d/15mqLmauipN3-Irw1cqo6wUBugOb7ZeTcERXpfilyCkg/edit?usp=sharing) is out.

### 4 November
* doc redesign - [user feedback](https://docs.google.com/document/d/1CVC-4eSbiRGIgH_FnDceyW0RklW7lYSxUOVOYZE-Shc/edit?usp=sharing), [review of current content](https://docs.google.com/document/d/1BYGzgdy0AlEUxf9-SQg4pJepLtPKXEdSO_mrjmHv7N4/edit?usp=sharing)

### 28 October
* we finished the [Staking](https://docs.google.com/document/d/1LXqvJ9O-vqt_8zpytkOT_EHyzjOVfU0GiwY0bpmr4Fc/edit?usp=sharing) and [ZKML](https://docs.google.com/document/d/1pICMATmT0rj_0rIik4FwHeRFPu8CU8RtndN5aI3gYkk/edit?usp=sharing) articles last week
* working on a complete update/redesign of the Mina Docs website. Now [reviewing current content](https://docs.google.com/document/d/1BYGzgdy0AlEUxf9-SQg4pJepLtPKXEdSO_mrjmHv7N4/edit?usp=sharing), launching [survey](https://docs.google.com/document/d/1f395D9MhM0sj1QWjalnONsSHuQrHCD1Z8Hv9cjylXkI/edit?usp=sharing) today.

### 21 october
* Finished [staking blog rewrite](https://docs.google.com/document/d/1LXqvJ9O-vqt_8zpytkOT_EHyzjOVfU0GiwY0bpmr4Fc/edit?usp=sharing)
* documentation redesign [survey](https://docs.google.com/document/d/1f395D9MhM0sj1QWjalnONsSHuQrHCD1Z8Hv9cjylXkI/edit?usp=sharing) and research 

### 14th October
* [Staking documnetation rewrite](https://docs.google.com/document/d/1WDC2VbYtJHHgOIe8iFCj7tmQoLuDXmaK9f78sH1B2eo/edit?usp=sharing)
* [sitemap of current documentation](https://miro.com/welcomeonboard/OVBFV3ZlZFBkOFExZFQ4bzVHY002VWhtMUpTUTdWVFZTSEFENDVzSVcxZEJjMDd1SkY4bHdzSG5hNTJwaUl4ZXwzNDU4NzY0NjAxODA0ODU5NzYzfDI=?share_link_id=238585188700)
* [survey for users regarding documentation re-design](https://mdvpmchvuew.typeform.com/to/HiEgsGBA)

### 7th October

* [Documentation website redesign - sitemap](https://miro.com/welcomeonboard/OVBFV3ZlZFBkOFExZFQ4bzVHY002VWhtMUpTUTdWVFZTSEFENDVzSVcxZEJjMDd1SkY4bHdzSG5hNTJwaUl4ZXwzNDU4NzY0NjAxODA0ODU5NzYzfDI=?share_link_id=59962124216)
* [Website redesign - sitemap](https://miro.com/app/board/uXjVLcv3l2c=/)
* [ZKML case study](https://docs.google.com/document/d/1uAinoNqXDqk9EtOb2O26n_fHubgKJu3rZQ1awWGM2wM/edit?usp=sharing)
* [Staking article on main web update](https://docs.google.com/document/d/1WDC2VbYtJHHgOIe8iFCj7tmQoLuDXmaK9f78sH1B2eo/edit?usp=sharing)

### 30th September
* [ZKML case study](https://docs.google.com/document/d/1uAinoNqXDqk9EtOb2O26n_fHubgKJu3rZQ1awWGM2wM/edit?usp=sharing)

### 23rd September
* [Why do we need ZKMLs?](https://docs.google.com/document/d/1VKd9cMJbI5qhUsRs56fMyOH9dlTvhOO2OYTm_D3_eSA/edit?usp=sharing)
* [Rust node intro article](https://docs.google.com/document/d/1jqI0E55U7ykYxhBJmnrLojjsaeT9TeBEtgXf911MLww/edit?usp=sharing)
* [Architecture overview update for Rust node](https://docs.google.com/document/d/1aJ1UWjCn0kAZ78AqU0odLDOPdC03lFa_M_Fi6FjN2mQ/edit?usp=sharing)
* [Ads on X - legal regulatory environments for APAC countries](https://docs.google.com/document/d/1qNaCe0UB-Y-c8mc2PkVafoLLmbjecbG58RP1DGswHDg/edit?usp=sharing)

### 16th September

* [Ads on X](https://docs.google.com/document/d/1qNaCe0UB-Y-c8mc2PkVafoLLmbjecbG58RP1DGswHDg/edit?usp=sharing)

### 9th September

* [Block production FE documentation](https://docs.google.com/document/d/1oVb8NbgtFrWW1ZyemANs8edvIXVLFDUBR-KAxFJ0_Vw/edit?usp=sharing)

### 26th August 

* Continued working on the ZKP documentation  - [SNARKs vs STARKs vs Bulletproofs comparison and use-cases.](https://docs.google.com/document/d/1DOWgUkGGUeMcBKFuoBbRL9D07wikTSdKoM2yxPbtaqo/edit?usp=sharing)
* did some work on the [email re-engagement campaign](https://docs.google.com/document/d/1MGEGNUJEfqrVXxaR_2LqGsWRZnPdRVHvOwqfsiBonI8/edit?usp=sharing) - but as it is on hold I won't do any more until we get a go-ahead.
* I've been contacted again by colossus, the fireblocks-like staking service, regarding cooperation with Mina, requested that they provide numbers - i.e. how many Mina are their potential customers willing to stake etc. 
* Had a call with Reddit ads regarding the PPCs - 5k min. investment and needs legal docs - but this is on hold until further notice.
* Had an intro call with Elizabeth McFaul new Product Marketing manager.

### 19th August

* Working on the onboarding ZKP documentation
*  Wrote [ZKP properties (soundness, completeness, zero-knowledge)](https://docs.google.com/document/d/1_rx2EQUe0_N-T_SivD3DQhss-iSx8mkzL196Ka0c6u0/edit?usp=sharing)
*  Wrote [SNARKs/STARKs/Bulletproofs](https://docs.google.com/document/d/1DOWgUkGGUeMcBKFuoBbRL9D07wikTSdKoM2yxPbtaqo/edit?usp=sharing)
* Had a call with Reddit marketing - they will allow ads for Mina as long as its an educational platform for ZKs, but we would need to rebrand it i.e. can't have it under Mina protocol main website.

### 12th August
* I am working on the ["Why do we need ZKPs"](https://docs.google.com/document/d/1Uat6K5Vvn8rGBRKarMiiwHqWN30-h31KlchH7p7aHt0/edit?usp=sharing) which will be the first article for the overall ZKP onboarding series that I am working on.
* Here is an overview of the entire [ZKP onboarding series.](https://docs.google.com/document/d/1xzI6weuY_97PwM8wBDvRygzeLwlyqSqx5rCdiaTKfVY/edit?usp=sharing)
* Last week I had a call with Andrea regarding the educational platform for ZKPs on the Mina website. There will be an upcoming re-design of the website and if it gets approved by Salah I will be working with the web designer to include this new platform. THe key idea is that through an educational platform, we can promote Mina on major social networks (reddit, FB, X.com), because otherwise ads for crypto services and products are banned.
* Last Friday I also worked on the readme for the new release with Bruno and Lukas.

### 29th July

* I am working on the ZKP onboarding series
* Currently writing the ["Why" part of ZKP](https://docs.google.com/document/d/1Uat6K5Vvn8rGBRKarMiiwHqWN30-h31KlchH7p7aHt0/edit?usp=sharing) - why do we need them, what happens if we don't have them etc.
* Also working on What part as it comes organically while working on the "why".
* I'm also looking into platforms for PPC campaigns for Mina - If we angle the marketing of Mina Protocol towards education, it would permit us to advertise on Twitter and Reddit. 
* Business development-wise, no replies from any contacted blockchains in last 2 weeks. Seems like things have slowed down for summer.
 
### 22nd July 

* As discussed with Andrea, I am working on an [ZKP onboarding series](https://docs.google.com/document/d/1xzI6weuY_97PwM8wBDvRygzeLwlyqSqx5rCdiaTKfVY/edit?usp=sharing) of existing and new docs.
* last week I was designing the general outline of onboarding series and locating existing files.
* I am also dividing my articles into small docs that can serve as steps in the onboarding process.

### 15th January

* There were not many updates to testing, so I spent most of last week converting and creating documentation:

**for planned changes (Q1/Q2):**
* [Witness folding](https://github.com/JanSlobodnik/pre-publishing/blob/main/witness-folding.md)

**for node's current use cases:**
* [Bootstrap & Catchup](https://github.com/JanSlobodnik/pre-publishing/blob/main/bootstrap-catchup.md)
* [Web Node - how it works](https://github.com/JanSlobodnik/pre-publishing/blob/main/web-node.md)

I also expanded:

* [Why we're developing the web node](https://github.com/JanSlobodnik/pre-publishing/blob/main/why-openmina.md)

All of these are in my own repo currently, will move to main once the docs are finalized and approved.

### 8th January

* Before release - Updated readme with [an overview of our documentation.](https://github.com/openmina/openmina). This includes the Testing Framework doc.
* Over the Christmas break I began [studying Rust](https://docs.google.com/document/d/1i7ynnZW1rL3qEOYA1h-pTL6KJKoyjDNPpJEMxUl1-Mg/edit) from a "high level" point of view, i.e. I want to understand how this programming language works, what are its advantages and limitations, how we can use it etc.

**Next tasks:**

* [Testing documentation](https://github.com/openmina/openmina/blob/main/docs/testing/testing.md) - This week I plan on improving the testing documentation, particularly, we need to add steps where applicable, and either remove all diagrams or add more diagrams (currently only one test has a series of diagrams).
* Improve existing documentation, now working on [the "Why" doc](https://github.com/JanSlobodnik/pre-publishing/blob/main/why-openmina.md).

**Optional tasks:**

* Work on future github documentation, including [Block Production](https://github.com/JanSlobodnik/pre-publishing/blob/main/block-production.md) and [VRF.](https://github.com/openmina/openmina/blob/feat/block_producer/vrf_evaluator/vrf/README.md)
* Figure out a better front end for our documentation (GitHub pages?)

### 18th December

* [Main testing documentation](https://github.com/openmina/openmina/blob/develop/testing.md) - removed some items we have not finished yet.
* As a user I'd like to see tooltips for values/categories in the UI, so I created a spreadsheet with copy for the [Dashboard and Network pages](https://docs.google.com/spreadsheets/d/1k1ezjiSTzMgxG_uUYnqO48aQ1Yn_qqjIVHYrn2GpfZo/edit#gid=0) (tooltips). 

Next Tasks:

* Add steps for tests where applicable
* While working on copy for the UI I found that we have a lot of [unpublished information on the tracing interface](https://docs.google.com/document/d/1AFaAF4wBS_JhnRZiru4xsSWTJ_ef2BALfeYqtUDUhb8/edit#heading=h.sao5ivrcvuhc), I want to check whether all the information and screenshots are up to date and then add it in the form of documentation to GitHub. 
* We have documentation on GitHub, but it is quite difficult to navigate between the individual markdown files. If there are no other tasks right now, I'd look into creating something via GitHub pages, basically a Table of Contents for all of our documentation akin to what we had previously with docs.tezedge.com

### 11th December

* [Main testing documentation](https://github.com/openmina/openmina/blob/develop/testing.md): created a table of contents for better navigation, added links to tests, added descriptions for some new tests. Reduced intro length, removed redundancies. I suggest we do not mention the Front End until release is ready.
* read the checklist for test scenarios. Fixed a few mistakes and formatting, added some [suggestions for this document](https://github.com/openmina/openmina/pull/150#issuecomment-1849110137), and will add more testing steps as they are completed. 
* Had some questions for Alex which were answered, and I've implemented the relevant information into the [main testing framework doc](https://github.com/openmina/openmina/blob/develop/testing.md).

**Next Tasks:**

* Potentially we could add additional diagrams for more complex tests, for example, those include three types of nodes (OCaml, Rust and OCaml/Rust seed nodes). 

### 4th December

Created readme w/ diagrams for Vlad's peer discovery tests:

https://github.com/openmina/openmina/blob/develop/node/testing/peer-discovery-test.md

Edited Alexander's test documentation:

https://github.com/openmina/openmina/blob/develop/node/testing/src/scenarios/multi_node/connection_discovery.md

Converted the testing framework overview doc into markdown and added it to our GitHub:

https://github.com/openmina/openmina/blob/develop/testing.md

Next tasks:

* Cleaning up the Open Mina testing framework overview.

* O1JS support for Web Node RFC conversion to pull request (with Teo).

* Potentially add diagrams by Lukas to the RFC

* VRF documentation (with Adrian).
 

### 27th November

**Main task - testing documentation**

Peer discovery test with diagrams:

https://github.com/openmina/openmina/blob/test/daily-discovery/docs/testing/discovery.md

OCaml - Rust node peer discovery (still in progress, once it is finalized I will add it to GitHub)

https://docs.google.com/document/d/16JJd3u_CtBWxRizvQmXYBVBSPlgrAgoEK51BVSyEBvg/edit

**Misc things:**

Began working on VRF implementation readme with Adrian:

https://github.com/openmina/openmina/blob/feat/block_producer/vrf_evaluator/vrf/README.md

Added intro + proofread Daniel's document on P2P overview:

https://github.com/openmina/openmina/blob/documentation/docs/p2p_service.md

Once UI is ready, will update the RFC for o1js support for Mina:

https://docs.google.com/document/d/1iUxL4cw-EOuHTC6bs1oONcEbZgVBkn7KP8_VuVQylGw/edit

### 20th November

**Updated the Network connectivity test:**

https://docs.google.com/document/d/1VfZH2vPR5wfGxvOqsOdHwWe7hbs5OLhhb6AQIemlVac/edit#heading=h.f4yxc8av5gcj

**Created a separate document for KAD testing:**

https://docs.google.com/document/d/1tZMbU0whTZTSblgcj-eKzbzLP1w4Emtd-MdDPcm0bhg/edit

this will undergo more changes before its added to the testing framework doc. One thing we need to clear up are the issues, whether they are fixable or not.

### 13th November

Updated some details about network connectivity tests, specifically why the discovery test is not working and pointing out some likely culprits. I'm waiting on other tests.

Since updates on the Testing Framework were sparse this week, together with Zura and Teo I began writing an RFC for o1 js support for the Mina Web node:

https://docs.google.com/document/d/1iUxL4cw-EOuHTC6bs1oONcEbZgVBkn7KP8_VuVQylGw/edit#heading=h.gdzpwlj6ll9d 

### 6th November

Added and described new tests - network connectivity. Still waiting for other tests.

### 30th October

Added screenshots, descriptions for screenshots, but they are placeholders so they will definitely change completely. Really last week I was trying to zoom out and get a big picture of what we are going for with the testing framework. I'm still waiting on the "How" part for many tests.

In the meantime I've proofread the readmes for some tests. - https://github.com/openmina/openmina/blob/develop/docs/testing/bootstrap.md

### 23rd October 

Expanded on each section, added documentation for node bootstrapping test. Waiting on other tests (the "How" parts)
